### PR TITLE
Add type enforcement to cvmfs::mount and cvmfs::domain

### PR DESCRIPTION
**Pull request by traylenator**
#### Pull Request (PR) description

Add type enforcement to `cvmfs::mount` and `cvmfs::domain`

In addition enforce all parameters are documented with puppet strings
using a linter.

Add puppet strings everywhere.

### CODE REVIEW

**Comment by traylenator** [comment]
Worth adding I plan to do a complete proof read of all docs before a next release.

File: manifests/domain.pp [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+# @param cvmfs_external_fallback_proxy Sets CVMFS_EXTERNAL_FALLBACK_PROXY
+# @param cvmfs_external_http_proxy Sets CVMFS_EXTERNAL_HTTP_PROXY
+# @param cvmfs_external_timeout Sets CVMFS_EXTERNAL_TIMEOUT
+# @param cvmfs_external_timeout_direct Sets CVMFS_EXTERNAL_TIMEOUT_DIRECT
+# @param cvmfs_external_url Sets CVMFS_EXTERNAL_URL
+# @param cvmfs_public_key Specify repository signing key
+# @param cvmfs_force_singing Force signing, deprecated.
+#
+define cvmfs::domain (
+  Optional[Integer] $cvmfs_quota_limit             = undef,
```
Comment (luisfdez) [inline manifests/domain.pp:25]: It seems that `cvmfs_quota_limit` is not used as is in `repo.local.erb`, it refers to `my_cvmfs_quota_limit` as passed from `config.pp`.

File: manifests/domain.pp [hunk 2]
```diff
@@ -1,21 +1,44 @@
...
+  Optional[String[1]] $cvmfs_server_url            = undef,
+  Optional[String[1]] $cvmfs_http_proxy            = undef,
+  Optional[Integer] $cvmfs_timeout                 = undef,
+  Optional[Integer] $cvmfs_timeout_direct          = undef,
+  Optional[Integer] $cvmfs_nfiles                  = undef,
+  Optional[String[1]] $cvmfs_public_key            = undef,
+  Optional[Stdlib::Yes_no] $cvmfs_force_singing    = undef,
```
Comment (luisfdez) [inline manifests/domain.pp:32]: Is this one used?

File: manifests/mount.pp [hunk 3]
```diff
@@ -1,30 +1,74 @@
...
+# @param cvmfs_external_fallback_proxy Sets CVMFS_EXTERNAL_FALLBACK_PROXY
+# @param cvmfs_external_http_proxy Sets CVMFS_EXTERNAL_HTTP_PROXY
+# @param cvmfs_external_timeout Sets CVMFS_EXTERNAL_TIMEOUT
+# @param cvmfs_external_timeout_direct Sets CVMFS_EXTERNAL_TIMEOUT_DIRECT
+# @param cvmfs_external_url Sets CVMFS_EXTERNAL_URL
+# @param cvmfs_repository_tag Sets CVMFS_REPOSITORY_TAG
+# @param mount_options Mount options to use for fstab style mounting.
+#
+define cvmfs::mount (
+  Optional[Integer]  $cvmfs_quota_limit                             = undef,
```
Comment (luisfdez) [inline manifests/mount.pp:47]: Same as before. It seems that we should apply the same transform to `cvmfs_quota_limit` as `config.pp` as the `erb` refers to `@my_cvmfs_quota_limit`.

File: manifests/mount.pp [hunk 4]
```diff
@@ -1,30 +1,74 @@
...
+  Optional[String[1]] $cvmfs_server_url                             = undef,
+  Optional[String[1]] $cvmfs_http_proxy                             = undef,
+  Optional[Integer] $cvmfs_timeout                                  = undef,
+  Optional[Integer] $cvmfs_timeout_direct                           = undef,
+  Optional[Integer] $cvmfs_nfiles                                   = undef,
+  Optional[String[1]] $cvmfs_public_key                             = undef,
+  Optional[Integer] $cvmfs_max_ttl                                  = undef,
+  Optional[Hash] $cvmfs_env_variables                               = undef,
+  Optional[Stdlib::Yes_no] $cvmfs_use_geoapi                        = undef,
+  Boolean$cvmfs_repo_list                                           = true,
```
Comment (luisfdez) [inline manifests/mount.pp:57]:
```suggestion
  Boolean $cvmfs_repo_list                                           = true,
```

**Review by luisfdez — CHANGES_REQUESTED** [review]

File: manifests/domain.pp (hunk 2 above)
Comment (traylenator) [reply manifests/domain.pp]: cvmfs_force_singing  is completly gone.

File: .sync.yml [hunk 5]
```diff
@@ -2,3 +2,8 @@
 spec/spec_helper_acceptance.rb:
   unmanaged: false
 
+Gemfile:
+  optional:
+    ':test:':
+      - gem: puppet-lint-param-docs
```
Comment (bastelfreak) [inline .sync.yml:8]: \o/

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**